OpenBSD declarations: virtual_set_stack_mapping(), and the OpenBSD fields
and helpers from emulators/wine.

Index: dlls/ntdll/unix/unix_private.h
--- dlls/ntdll/unix/unix_private.h.orig
+++ dlls/ntdll/unix/unix_private.h
@@ -256,6 +256,10 @@ extern void process_exit_wrapper( int st
 extern size_t server_init_process(void);
 extern void server_init_process_done(void);
 extern void server_init_thread( void *entry_point, BOOL *suspend );
+#ifdef __OpenBSD__
+extern void virtual_set_stack_mapping( void *start, SIZE_T size );
+extern BOOL virtual_is_readable_page( const void *addr );
+#endif
 extern int server_pipe( int fd[2] );
 
 extern void fpux_to_fpu( I386_FLOATING_SAVE_AREA *fpu, const XSAVE_FORMAT *fpux );
@@ -334,6 +338,21 @@ extern void virtual_fill_image_informati
 extern void *get_builtin_so_handle( void *module );
 extern NTSTATUS load_builtin_unixlib( void *module, const char *name );
 
+#ifdef __OpenBSD__
+/* code patching for OpenBSD cases */
+enum patch_result
+{
+    PATCH_DONE,         /* the code has now new bytes */
+    PATCH_CHANGED,      /* the code wasn't the expected old byte, another thread already updated it */
+    PATCH_FAILED        /* this code cannot be patched */
+};
+#define CODE_STUB_SLOT_SIZE 128 /* bytes per stub we generate for %gs patching, has to be power of 2 */
+extern enum patch_result virtual_patch_code( void *addr, const void *old_bytes, const void *new_bytes, SIZE_T size );
+extern void *virtual_alloc_code_stub( const void *near, const void *stub, SIZE_T size );
+extern void virtual_free_code_stub( void *slot );
+extern void *virtual_code_stub_slot( const void *addr );
+#endif
+
 extern NTSTATUS get_thread_ldt_entry( HANDLE handle, THREAD_DESCRIPTOR_INFORMATION *info, ULONG len );
 extern void *get_native_context( CONTEXT *context );
 extern void *get_wow_context( CONTEXT *context );
